### PR TITLE
Stop running tests in official builds

### PR DESCRIPTION
**Pull request by premun**
We shouldn't be running unit tests in official builds.

I created 4 scripts for clarity (PR/internalbuild.sh/cmd) and in case we would like to customize more in the future.

Fixes: https://github.com/dotnet/xharness/issues/58

### CODE REVIEW

File: azure-pipelines.yml [hunk 1]
```diff
@@ -63,23 +63,33 @@ stages:
...
-        - task: PublishTestResults@2
-          displayName: 'Publish Unit Test Results'
-          inputs:
-            testResultsFormat: xUnit
-            testResultsFiles: '$(Build.SourcesDirectory)/artifacts/TestResults/**/*.xml'
-            mergeTestResults: true
-            searchFolder: $(system.defaultworkingdirectory)
-            testRunTitle: XHarness unit tests - $(Agent.JobName)
-          condition: succeededOrFailed()
+        - ${{ if eq(variables._RunAsPublic, True) }}:
```
Comment (MattGal) [inline azure-pipelines.yml:83]: This doesn't need to be conditional, there just won't be test results if you don't make them.

File: azure-pipelines.yml [hunk 2]
```diff
@@ -63,23 +63,33 @@ stages:
...
                 _BuildConfig: Debug
         steps:
-        - script: eng\common\cibuild.cmd
-            -configuration $(_BuildConfig)
-            -prepareMachine
-            $(_InternalBuildArgs)
-          name: Build
-          displayName: Build
-          condition: succeeded()
+        - ${{ if eq(variables._RunAsPublic, False) }}:
```
Comment (MattGal) [inline azure-pipelines.yml:66]: I strongly prefer not having duplicate-except-for-running-tests scripts.  Why not just change the value of a variable like $(_InternalBuildArgs) conditionally?

File: azure-pipelines.yml (hunk 1 above)
Comment (premun) [reply azure-pipelines.yml:83]: It gives warnings though:
image

Not sure we want those on every build, so I added these guards

Comment (MattGal) [reply azure-pipelines.yml:83]: sure that's fine.  Less pressing than the "don't duplicate the whole script to remove 1 arg" issue :)

File: azure-pipelines.yml (hunk 2 above)
Comment (premun) [reply azure-pipelines.yml:66]: Good point, will do

**Comment by premun** [comment]
>The whole point of the %* and $@ characters in these scripts is to allow external arguments to come in, such as from Yaml. Please refactor this to call the same script with different args; this is too much duplication.

@MattGal can you be more specific? My problem was that `CIBuild.cmd` from Arcade commons is already calling with `-test`:
```
powershell -ExecutionPolicy ByPass -NoProfile -command "& """%~dp0Build.ps1""" -restore -build -test -sign -pack -publish -ci %*"
```

So whatever I append through `%*` won't help. Or there is a way? That is why I created the new cmd/sh.

**Comment by MattGal** [comment]
> > The whole point of the %* and $@ characters in these scripts is to allow external arguments to come in, such as from Yaml. Please refactor this to call the same script with different args; this is too much duplication.
> 
> @MattGal can you be more specific? My problem was that `CIBuild.cmd` from Arcade commons is already calling with `-test`:
> 
> ```
> powershell -ExecutionPolicy ByPass -NoProfile -command "& """%~dp0Build.ps1""" -restore -build -test -sign -pack -publish -ci %*"
> ```
> 
> So whatever I append through `%*` won't help. Or there is a way? That is why I created the new cmd/sh.
> 
> I guess I can bypass it and call the build.cmd directly..

No you can just trump it via Msbuild properties.  Example: https://github.com/dotnet/arcade/blob/592ec36c7fcfbd4c3a6025c61456e351aa2c9057/azure-pipelines.yml#L64

File: azure-pipelines.yml [hunk 3]
```diff
@@ -63,23 +63,33 @@ stages:
...
-            mergeTestResults: true
-            searchFolder: $(system.defaultworkingdirectory)
-            testRunTitle: XHarness unit tests - $(Agent.JobName)
-          condition: succeededOrFailed()
+        - ${{ if eq(variables._RunAsPublic, True) }}:
+          - task: PublishTestResults@2
+            displayName: 'Publish Unit Test Results'
+            inputs:
+              testResultsFormat: xUnit
+              testResultsFiles: '$(Build.SourcesDirectory)/artifacts/TestResults/**/*.xml'
```
Comment (mandel-macaque) [inline azure-pipelines.yml:88]: AFAIK, we have NUnit and xUnit tests atm. We should move https://github.com/dotnet/xharness/blob/master/tests/Microsoft.DotNet.XHarness.iOS.Tests/AppInstallerTests.cs to xUnit. I have created an issue for that: https://github.com/dotnet/xharness/issues/65

File: azure-pipelines.yml (hunk 2 above)
Comment (premun) [reply azure-pipelines.yml:66]: I clumped the `Build` and then `Build and run tests`+`Publish tests` together and used the override for the Test MSBuild param. I tested both the internal and public runs to verify everything

File: azure-pipelines.yml (hunk 3 above)
Comment (premun) [reply azure-pipelines.yml:89]: Good catch, forgot to migrate those as well. Fixed in #68

**Pull request merged**